test/spanner.ts

```diff
@@ -512,6 +513,109 @@ describe('Spanner with mock server', () => {
     });
   });
 
+  describe('session-not-found', () => {
+    it('should retry "Session not found" errors on Database.run()', done => {
+      const db = newTestDatabase({
+        min: 0,
+      });
+      spannerMock.setExecutionTime(
+        spannerMock.executeStreamingSql,
+        SimulatedExecutionTime.ofError({
+          code: status.NOT_FOUND,
+          message: 'Session not found',
+        } as MockError)
+      );
+      db.run(selectSql, (err, rows) => {
+        if (err) {
+          assert.fail(err);
+          done();
+          return;
+        }
+        assert.strictEqual(rows!.length, 3);
+        db.getSessions((err, results) => {
+          if (err) {
+            assert.fail(err);
+            done();
+            return;
+          }
+          // The server should have exactly 2 sessions. The first one that was
+          // created was removed from the session pool because of the simulated
+          // 'Session not found' error, but it was not removed from the server.
```

Comment:
"but it was not removed from the server" I don't quite understand this comment. The session was not removed from the server, why?

Reply:
We only simulate the `Session not found` error by telling the mock server to throw this error. The test case does not delete the session on the mock server, so if we check which sessions are still on the mock server, we will get both the original session and the session created by the retry. The first session is however removed from the session pool in the client, as the client thinks that the session is no longer valid.

I've tried to clarify the comment.